packages/redis/src/index.ts

```diff
@@ -0,0 +1,284 @@
+
+const vaultDepositSchema = jsonString.pipe(
+  z
+    .object({
+      amount: u128,
+      destination_address: z.string(),
+      input_asset: assetAndChain,
+      output_asset: assetAndChain,
+      deposit_chain_block_height: z.number().nullable().optional(),
+      affiliate_fees: z.array(accountFee),
+      broker_fee: accountFee.optional(),
+      max_boost_fee: z.number().optional(),
+      dca_params: z
+        .object({
+          chunk_interval: z.number(),
+          number_of_chunks: z.number(),
+        })
+        .nullable()
+        .optional(),
+      refund_params: z
+        .object({
+          min_price: u128,
+          retry_duration: z.number(),
+          refund_address: z.string(),
+        })
+        .nullable()
+        .optional(),
+      ccm_deposit_metadata: z
+        .object({
+          channel_metadata: z.object({
+            ccm_additional_data: z.any(),
+            message: z.string(),
+            gas_budget: z
+              .union([numericString, hexString])
+              .transform((n) => hexEncodeNumber(BigInt(n))),
+          }),
+        })
+        .nullable()
+        .optional(),
+    })
+    .transform(transformKeysToCamelCase),
+);
+
+type ChainBroadcast<C extends Exclude<ChainflipChain, 'Solana'>> = z.infer<
+  (typeof broadcastParsers)[C]
+>;
+
+type EthereumBroadcast = ChainBroadcast<'Ethereum'>;
+type PolkadotBroadcast = ChainBroadcast<'Polkadot'>;
+type BitcoinBroadcast = ChainBroadcast<'Bitcoin'>;
+type Broadcast = ChainBroadcast<Exclude<ChainflipChain, 'Solana'>>;
+
+const mempoolTransaction = jsonString.pipe(
+  z.object({
+    confirmations: z.number(),
+    value: u128,
+    tx_hash: z.string(),
+    deposit_chain_block_height: z.number().optional(),
+  }),
+);
+
+export default class RedisClient {
+  private client;
+
+  constructor(url: `redis://${string}` | `rediss://${string}`) {
+    this.client = new Redis(url);
+  }
+
+  async getBroadcast(
+    chain: 'Ethereum',
+    broadcastId: number | bigint,
+  ): Promise<EthereumBroadcast | null>;
+  async getBroadcast(
+    chain: 'Polkadot',
+    broadcastId: number | bigint,
+  ): Promise<PolkadotBroadcast | null>;
+  async getBroadcast(
+    chain: 'Bitcoin',
+    broadcastId: number | bigint,
+  ): Promise<BitcoinBroadcast | null>;
+  async getBroadcast(
+    chain: 'Arbitrum',
+    broadcastId: number | bigint,
+  ): Promise<EthereumBroadcast | null>;
+  async getBroadcast(
+    chain: ChainflipChain,
+    broadcastId: number | bigint,
+  ): Promise<Broadcast | null>;
+  async getBroadcast(
+    chain: ChainflipChain,
+    broadcastId: number | bigint,
+  ): Promise<Broadcast | null> {
+    if (chain === 'Solana') return null;
+    const key = `broadcast:${chain}:${broadcastId}`;
+    const value = await this.client.get(key);
+    return value ? broadcastParsers[chain].parse(JSON.parse(value)) : null;
+  }
+
+  async getDeposits(asset: ChainflipAsset, address: string): Promise<PendingDeposit[]> {
```

Comment:
i think this is the only change i made, i changed the signature from requiring a chain and asset to just an internal asset